app/section/sections/mobile-memory-and-storage/ui.test.js

```diff
@@ -0,0 +1,240 @@
+  await pageSetup(t);
+
+  const storageRequirementsDescription = Selector('[data-test-id="question-storage-requirements-description"]');
+
+  await t
+    .expect(storageRequirementsDescription.find('label.nhsuk-label').innerText).eql('Description of storage requirements')
+    .expect(storageRequirementsDescription.find('span.nhsuk-hint').innerText).eql('Add further information about the storage requirements needed for your Solution to function.')
+    .expect(storageRequirementsDescription.find('textarea').count).eql(1)
+    .expect(storageRequirementsDescription.find('[data-test-id="textarea-field-footer"]').innerText).eql('You can enter up to 300 characters');
+});
+
+test('should populate the storage requirements description with existing data', async (t) => {
+  pageSetup(t, true);
+
+  const theQuestions = Selector('form');
+
+  await t
+    .expect(theQuestions.find('[data-test-id="question-storage-requirements-description"]').find('textarea').value).eql('Some storage requirements description');
+});
+
+test('should show error summary and validation for storage requirements description indicating it is mandatory', async (t) => {
+  await pageSetup(t);
+
+  nock('http://localhost:8080')
+    .put('/api/v1/Solutions/S100000-001/sections/mobile-memory-and-storage')
+    .reply(400, {
+      required: ['storage-requirements-description'],
+    });
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const storageRequirementsDescription = Selector('[data-test-id="question-storage-requirements-description"]');
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+  await t
+    .expect(errorSummary.exists).notOk()
+    .click(submitButton.find('button'))
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql('Storage requirement description is a mandatory question')
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#storage-requirements-description')
+    .expect(storageRequirementsDescription.find('.nhsuk-textarea--error').exists).ok()
+    .expect(storageRequirementsDescription.find('.nhsuk-error-message').innerText).eql('Error:\nStorage requirement description is a mandatory question');
+});
+
+test('should show error summary and validation for storage requirements description question it exceeds the maxLength', async (t) => {
+  await pageSetup(t);
+
+  nock('http://localhost:8080')
+    .put('/api/v1/Solutions/S100000-001/sections/mobile-memory-and-storage')
+    .reply(400, {
+      maxLength: ['storage-requirements-description'],
+    });
+
+  const oneHundredCharacters = '0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789';
+  const threeHundredCharacters = oneHundredCharacters.repeat(3);
```

Comment:
I might have given you not a great one to copy from. But we shouldn't need this. Since we mock the response from the API you can remove the above 2 consts